services/command-router-base/src/main/java/org/eclipse/hono/commandrouter/impl/amqp/ProtonBasedCommandConsumerFactoryImpl.java

```diff
@@ -82,9 +83,10 @@ public ProtonBasedCommandConsumerFactoryImpl(
     }
 
     @Override
-    public void initialize(final CommandTargetMapper commandTargetMapper) {
+    public void initialize(final TenantClient tenantClient, final CommandTargetMapper commandTargetMapper) {
```

Comment:
can't we simply add *tenantClient* and *commandTargetMapper* to the constructor and get rid of this method? We already have life cycle methods and having an additional *initialize* before we can actually start the component seems really awkward to me ...

Reply:
I've refactored the corresponding classes now to remove the `initialize` methods, also removing `@Autowired` annotations in `CommandRouterServiceImpl` (in the command router base module).